static/js/AddToSourceSheet.jsx

```diff
@@ -120,11 +120,32 @@ class AddToSourceSheetBox extends Component {
           source[language.slice(0,2)] = selectedWords;
         }
       }
-      let postData = {source: JSON.stringify(source)};
-      if (this.props.note) {
-        postData.note = this.props.note;
+      if (this.validateRefs(source.refs)) {
+        let postData = {source: JSON.stringify(source)};
+        if (this.props.note) {
+          postData.note = this.props.note;
+        }
+        $.post(url, postData, this.confirmAdd);
+      }
+  }
+  validateRefs(refs) {
+    // validate refs have no images before posting them to sheet
+    let foundImage = false;
+    refs.forEach(r => {
```

Comment:
@stevekaplan123 this code can be simpler with a standard for-loop. Once an image is found, call the alert and break out of the loop. This should remove two if-statements from the code.